src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -355,8 +355,11 @@ object DaffodilPlugin extends AutoPlugin {
           "<=3.10.0" -> "2.12"
```

Comment:
Above we havea  comment that says "the daffodil-japi dependency ....". Since it could be daffodil-core, we should update it to something like "the Daffodil dependency needed for daffodilPackageBin must ignore the scalaVersion ..."